modules/account-lib/src/coin/eth2/keyPair.ts

```diff
@@ -0,0 +1,57 @@
+import * as BLS from '@chainsafe/bls';
+import { addHexPrefix, pubToAddress } from 'ethereumjs-util';
+import { DefaultKeys, isPrivateKey, KeyPairOptions } from '../baseCoin/iface';
+import { BlsKeyPair } from '../baseCoin/blsKeyPair';
+
+/**
+ * Ethereum keys and address management.
+ */
+export class KeyPair extends BlsKeyPair {
+  /**
+   * Public constructor. By default, creates a key pair with a random master seed.
+   *
+   * @param { KeyPairOptions } source Either a master seed, a private key (extended or raw), or a public key
+   *     (extended, compressed, or uncompressed)
+   */
+  constructor(source?: KeyPairOptions) {
+    super(source);
+    if (source) {
+      // TODO: FIX THIS CHECK
+      if (isPrivateKey(source)) {
+        super.recordKeysFromPrivateKey(source.prv);
+      } else {
+        throw new Error('Invalid key pair options');
+      }
+    } else {
+      throw new Error('Invalid key pair options');
+    }
+  }
+
+  /**
+   * ETH2 default keys format is a pair of Uint8Array keys
+   *
+   * @returns { DefaultKeys } The keys in the defined format
+   */
+  getKeys(): DefaultKeys {
+    const result: DefaultKeys = {
+      // Need some way to get pubkey from keyPair item
+      pub: BLS.PublicKey.fromBytes(this.keyPair.pub).toHexString(),
+    };
+
+    if (this.keyPair.prv) {
+      // Need some way to get pubkey from keyPair item
+      result.prv = BLS.PrivateKey.fromBytes(this.keyPair.prv).toHexString();
+    }
+    return result;
+  }
+
+  /**
+   * Get an Ethereum public address
+   *
+   * @returns {string} The address derived from the public key
+   */
+  getAddress(): string {
+    const publicKey = Buffer.from(this.getKeys().pub, 'hex'); // first two characters identify a public key
+    return addHexPrefix(pubToAddress(publicKey, true).toString('hex'));
```

Comment:
This is an ETH1 library and not how addresses are generated in eth2. In fact I'm surprised this works at all since the pubkey format is different as well